lib/cpp/CMakeLists.txt

```diff
@@ -26,7 +26,7 @@ if(WITH_BOOSTTHREADS)
 else()
   find_package(Boost 1.53.0 REQUIRED)
 endif()
-include_directories(SYSTEM "${Boost_INCLUDE_DIRS}")
```

Comment:
SYSTEM implies -isystem for gcc/clang thus can avoid polluting log with warnings from boost.